server/src/main/java/org/elasticsearch/index/mapper/IpFieldMapper.java

```diff
@@ -497,18 +499,18 @@ private static InetAddress value(XContentParser parser, InetAddress nullValue) t
     }
 
     private void indexValue(DocumentParserContext context, InetAddress address) {
+        if (dimension) {
+            try {
+                // Extract the tsid part of the dimension field
+                BytesReference bytes = TimeSeriesIdFieldMapper.extractTsidValue(NetworkAddress.format(address));
```

Comment:
I thought about it too, but I preferred to avoid having too many `extractTsidValue()` `static` methods. I know that for now we only have 3 dimension field types (`keyword`, `ip, `number`) but maybe we add more in the future (geo stuff). Besides, I thought it is the `IpFieldMapper''s responsibility to serialize and `address` to a `String` (similarly the `NumberFieldMapper` serializes any number type to a `long`). 

In any case, I think this is a matter of style. Happy to change it if you think it can be useful